python/deltalake/writer.py

```diff
@@ -202,6 +206,22 @@ def visitor(written_file: Any) -> None:
         )
 
 
+def __throw_error_if_delta_append_only_and_mode_not_append(
+    table: Optional[DeltaTable],
+    configuration: Optional[Mapping[str, Optional[str]]],
+    mode: str,
+) -> None:
+    if table:
+        configuration = table.metadata().configuration
+    config_delta_append_only = (
+        configuration and configuration.get("delta.appendOnly", "false") == "true"
+    )
+    if config_delta_append_only and mode != "append":
+        raise ValueError(
+            "If configuration has delta.appendOnly = 'true', mode must be 'append'"
```

Comment:
**nitpick:** add the mode in the `ValueError` to get the current value.

Reply:
Thanks for the callout! This is a great point to help people who might cross this error.